openxr/src/vive_tracker_paths.rs

```diff
@@ -0,0 +1,20 @@
+use crate::Path;
+
+pub struct ViveTrackerPathsHTCX {
+    pub persistent: Path,
+    pub role: Option<Path>,
+}
+
+impl From<*mut sys::ViveTrackerPathsHTCX> for ViveTrackerPathsHTCX {
+    fn from(paths: *mut sys::ViveTrackerPathsHTCX) -> Self {
+        let paths = unsafe { *paths };
```

Comment:
This is unsound; safe code could pass in a null pointer. Perform the deref in the generated code instead.

Reply:
Ah good point. Couldn't a user build the wrapper type themselves with a null `paths` anyways? Should we change the return type to `Option` and return `None` when it's null instead?

Reply:
I'm not sure what you mean. If the user derefs a null pointer, they've invoked UB. We should not provide safe interfaces that permit this to occur. Null `Path`s are not null pointers, and are fortunately much safer to handle.

Reply:
I'm specifically talking about [this field](https://github.com/Ralith/openxrs/pull/97/files#diff-2d9ed58d2f856284b7dc54da37b3e9690c24a83d3d15358961e0a2cce8bdea96R5358) being a null pointer. In that case, wouldn't a deref in the [generated code](https://github.com/Ralith/openxrs/pull/97/files#diff-9302c9ce141dbd6e8d9a2fa27588e9701a01ec8f3eb53b52e1ccb9676cb3a189R1644) still be UB?

Reply:
Neither of those links address a specific line of code, so I'm not sure what you're referring to.

Reply:
Oh 🤦 looks like links to lines in review diffs don't work.

I'm specifically talking about [this field](https://github.com/Ralith/openxrs/blob/b287335bbdd253312d1db20faf119e445f80a997/sys/src/generated.rs#L5358) being a null pointer. In that case, wouldn't a deref in the [generated code](https://github.com/Ralith/openxrs/blob/b287335bbdd253312d1db20faf119e445f80a997/openxr/src/generated.rs#L1644) still be UB?

Reply:
Ah, yes. Is this really the first time we've needed to traverse a pointer in an event? Huh. We should probably make `ViveTrackerConnectedHTCX::new` unsafe, with the requirement that the argument must satisfy the requirements of the OpenXR spec placed upon such values when delivered by OpenXR.